app/code/Bulbulatory/Recommendations/Controller/Customer/SendRecommendation.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Bulbulatory\Recommendations\Controller\Customer;
+
+use Exception;
+use Throwable;
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Bulbulatory\Recommendations\Model\RecommendationRepository;
+use Psr\Log\LoggerInterface;
+use Magento\Customer\Model\Session;
+use Bulbulatory\Recommendations\Helper\Email;
+
+class SendRecommendation extends Action
+{
+    /**
+     * @var RecommendationRepository
+     */
+    private $recommendationsRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var Session
+     */
+    private $customerSession;
+
+    /**
+     * @var string
+     */
+    private $email;
+
+    /**
+     * @var
+     */
+    private $hash;
+
+    /**
+     * @var Email
+     */
+    private $emailHelper;
+
+    /**
+     * SendRecommendation constructor.
+     * @param Context $context
+     * @param RecommendationRepository $recommendationRepository
+     * @param LoggerInterface $logger
+     * @param Session $customerSession
+     * @param Email $emailHelper
+     */
+    public function __construct(
+        Context $context,
+        RecommendationRepository $recommendationRepository,
+        LoggerInterface $logger,
+        Session $customerSession,
+        Email $emailHelper
+    )
+    {
+        $this->recommendationsRepository = $recommendationRepository;
+
+        $this->logger = $logger;
+
+        $this->customerSession = $customerSession;
+
+        $this->emailHelper = $emailHelper;
+
+        parent::__construct($context);
+    }
+
+    /**
+     * @return mixed
+     */
+    public function execute()
+    {
+        try {
+            $this->setEmail($this->getRequest()->getParam('email'));
+            $this->saveRecommendation();
+            $this->sendRecommendation();
+
+            $this->messageManager->addSuccessMessage(__('Recommendation send to email: %1', $this->email));
+        }
+        catch (Throwable $exception)
+        {
+            $this->messageManager->addErrorMessage(__($exception->getMessage()));

```

Comment:
Same as above, about displaying exception messages to the customer.

Reply:
@TM18

> Same as above, about displaying exception messages to the customer.

Changed!